js/client-battle.js

```diff
@@ -379,7 +379,7 @@
 					if (this.request.forceSwitch !== true) {
 						var faintedLength = _.filter(this.request.forceSwitch, function (fainted) {return fainted;}).length;
 						var freedomDegrees = faintedLength - _.filter(switchables.slice(this.battle.mySide.active.length), function (mon) {return !mon.fainted;}).length;
-						this.choice.freedomDegrees = Math.max(freedomDegrees, 0);
+						this.choice.freedomDegrees = Math.min(Math.max(freedomDegrees, 0), this.battle.pokemonControlled);
```

Comment:
Whether the Pokémon belongs to the player should be checked in the ``_.filter()`` above instead.